ggircs-app/app/components/XmlDiff/ReportSelector.tsx

```diff
@@ -1,50 +1,47 @@
-import { useState, useCallback } from "react";
+import { useState } from "react";
 import { RelayReportObject } from "types";
 import Input from "@button-inc/bcgov-theme/Input";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
 import debounce from "lodash.debounce";
-import { useRouter } from "next/router";
+import { NextRouter } from "next/router";
 
 interface Props {
   diffSide: String;
   allReports: readonly RelayReportObject[];
+  router: NextRouter;
 }
 
 export const ReportSelector: React.FunctionComponent<Props> = ({
   diffSide,
   allReports,
+  router,
 }) => {
-  const router = useRouter();
-
```

Comment:
I'm a bit confused as to why this is needed, the router should be a singleton, no?

Reply:
it's *almost* a singleton - it uses the `React.Context` API ([code](https://github.com/vercel/next.js/blob/00a843280d340862dabcbf96248acf95c0957627/packages/next/client/router.ts#L137)), that creates a singleton per react tree from what I understand, so if we have 2 branches using the router we aren't guaranteed uniqueness and things get murky. @dleard  and I ran into that issue before.